### PR TITLE
fixes for refresher includes configuration system data - template ins…

### PR DESCRIPTION
**Pull request by blancett**
…tances

### CODE REVIEW

**Comment by blancett** [comment]
will fix the conflicts

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb [hunk 1]
```diff
@@ -27,17 +27,18 @@ def configuration_profile_inv_to_hashes(profiles)
...
           {
             :type         => "ManageIQ::Providers::CloudAutomationManager::ConfigurationManager::ConfiguredSystem",
             :manager_ref  => cs["id"].to_s,
             :hostname     => cs["name"],
-            :last_checkin => cs["last_compile"],
-            :build_state  => cs["build"] ? "pending" : nil,
-            :ipaddress    => cs["ip"],
-            :mac_address  => cs["mac"],
-            :ipmi_present => cs["sp_ip"].present?,
+            :last_checkin => cs["created_at"],
```
Comment (bdunne) [inline app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb]: For Foreman this is the last time a system checked in with the puppet master.  Is this a one time interaction, or is there a better field to look at for recurring interactions between the server and the configured system?

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb [hunk 2]
```diff
@@ -41,13 +50,14 @@ def collect_configuration_inventory(connection, base_url)
...
-        # get all templates
-        template_uri_str = base_url + ":30000/cam/api/v1/templates?tenantId=" + tenant_id + "&ace_orgGuid=" + all + "&cloudOE_spaceGuid=" + team
-        # get stacks
-        #template_uri_str = base_url + ":30000/cam/api/v1/stacks?tenantId=" + tenant_id + "&ace_orgGuid=" + all + "&cloudOE_spaceGuid=" + team
+        # url to get all template resources
+        template_uri_str = base_url + api_url + "?tenantId=" + tenant_id + "&ace_orgGuid=" + all + "&cloudOE_spaceGuid=" + team
         template_body = redirect_cam_api(template_uri_str, 5, connection)
-        result[:templates] = JSON.parse(template_body.body)
-
+        if type == "templates"
```
Comment (bdunne) [inline app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb]: I would probably put this logic on the caller side and return `JSON.parse(template_body.body)`

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb [hunk 3]
```diff
@@ -4,8 +4,16 @@ module CloudAutomationManager
     class ConfigurationManager::Refresher < ManageIQ::Providers::BaseManager::Refresher
       def parse_legacy_inventory(manager)
         manager.with_provider_connection do |connection|
-          data = collect_configuration_inventory(connection, manager.url)
+          # get templates
```
Comment (bdunne) [inline app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb]: Then the caller side can look like this:
```
data = {}

# get templates
template_url_str = ":30000/cam/api/v1/templates"
data[:templates] = collect_configuration_inventory(connection, manager.url, template_url_str)

# get configured systems
stack_url_str = ":30000/cam/api/v1/stacks"
data[:stacks] = collect_configuration_inventory(connection, manager.url, stack_url_str)

ConfigurationManager::RefreshParser.configuration_inv_to_hashes(data)
```

**Comment by bdunne** [comment]
Also, can you add to the existing `refresher_spec.rb`...

```ruby
 def assert_ems_counts
  expect(ems.configuration_profiles.count).to eq(169)
  expect(ems.configured_systems.count).to eq(the number you have in your test system)
end
```

and a new method named `assert_specific_configured_system` testing the attributes of a specific configured system similar to the logic in `assert_specific_configuration_profile` and call that just after `assert_specific_configuration_profile` on line 28.

In order to get the test to pass, you'll probably need to delete the existing VCR cassette then run the test (which will record a new one).

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb (hunk 1 above)
Comment (blancett) [reply app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb]: In terraform you can do updates to deployed systems - the created date will change based on updates. There is a lot more information we want to capture and I am trying to hammer some of that data in the configured system schema you are using. We will definitely need to revisit it.

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb (hunk 3 above)
Comment (blancett) [reply app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb]: Yeah good call - that is cleaner.

File: .project [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (blancett) [reply .project]: yep - think my git add was a little aggressive. Let me see if i can fix these. Been struggling a bit with git.

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb [hunk 5]
```diff
@@ -27,17 +27,18 @@ def configuration_profile_inv_to_hashes(profiles)
...
+            :build_state  => cs["build"],
+            :ipaddress    => cs["templateId"],
+            :mac_address  => cs["status"],
+            :ipmi_present => cs["templateName"]
```
Comment (agrare) [inline app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb]: @blancett are these accurate?  IP Address is stored in `templateId`, MAC Address in `status`?

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb (hunk 3 above)
Comment (agrare) [reply app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb]: `collect_configuration_inventory` is expected to return a hash with all of the collected data

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb [hunk 6]
```diff
@@ -4,8 +4,16 @@ module CloudAutomationManager
...
+          template_url_str = ":30000/cam/api/v1/templates"
+          type = "template"
+          data = collect_configuration_inventory(connection, manager.url, template_url_str, type)
           ConfigurationManager::RefreshParser.configuration_inv_to_hashes(data)
+          # get configured systems
+          stack_url_str = ":30000/cam/api/v1/stacks"
+          type = "stacks"
+          data_systems = collect_configuration_inventory(connection, manager.url, stack_url_str, type)
+          ConfigurationManager::RefreshParser.configuration_inv_to_hashes(data_systems)
```
Comment (agrare) [inline app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresher.rb]: And we should only be calling `configuration_inv_to_hashes` once with all of the collected data

File: app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb (hunk 5 above)
Comment (blancett) [reply app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb]: yeah i am just cramming in data - not of this makes sense for terraform. If they show up in the table we are good. We will revisit all this after the POC.

Comment (blancett) [reply app/models/manageiq/providers/cloud_automation_manager/configuration_manager/refresh_parser.rb]: yes just cramming in data for the ui table - not of it matches - we will have to revisit after the POC.

**Comment by agrare** [comment]
@blancett do you want some help resolving the conflicts?

**Pull request merged**